src/wp-includes/default-filters.php

```diff
@@ -346,8 +346,8 @@
 add_action( 'init', '_register_core_block_patterns_and_categories' );
 add_action( 'init', 'check_theme_switched', 99 );
 add_action( 'init', array( 'WP_Block_Supports', 'init' ), 22 );
-add_action( 'switch_theme', array( 'WP_Theme_JSON_Resolver', 'clean_cached_data' ) );
-add_action( 'start_previewing_theme', array( 'WP_Theme_JSON_Resolver', 'clean_cached_data' ) );
+add_action( 'switch_theme', '_wp_clean_theme_json_caches' );
```

Comment:
If https://github.com/WordPress/wordpress-develop/pull/3712 lands first, these can be ignored, as they'd have been already ported over there.

Reply:
Hmm, this change seems out-of-scope for the purpose of this PR. What is the impact of committing this PR without this change and before PR #3712 is committed?

Reply:
responded at https://github.com/WordPress/wordpress-develop/pull/3556#issuecomment-1361689398

Reply:
IMO keeping this in here is okay. It's not as critical to clean caches for now, given that the cache is non-persistent, but it's still a good practice which we will be able to rely on more in the future. I also think having the `_wp_clean_theme_json_caches()` function itself makes sense, since we also introduce the `wp_theme_has_theme_json()` function, which uses that cache.